### PR TITLE
Remove Source Link package references

### PR DESCRIPTION
**Pull request by tmat**
Source Link is now included in .NET SDK.

### CODE REVIEW

**Comment by tmat** [comment]
@mmitche

**Review by ViktorHofer — CHANGES_REQUESTED** [review]
I'm excited to see sourcelink support added directly into the SDK.

We must wait until Arcade updates to an SDK that has the implicit sourcelink support though. Based on https://github.com/dotnet/sdk/commit/dd2e7b5c8a3e3c8e4fc4292d1d97d135195bd490, that would be 8.0 Preview 4.

We should also communicate this change with the Arcade Engineering Partners alias as this is a breaking change. Before, you could set an explicit source link package version and Arcade would have respected that while now such an explicit version would just be ignored.

**Comment by mmitche** [comment]
Yep, this should not be merged until p4 is released and arcade updates to P4.

**Review by mmitche — DISMISSED** [review]
Wait till p4 release

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -56,15 +56,6 @@
       <Uri>https://github.com/dotnet/linker</Uri>
       <Sha>3efd231da430baa0fd670e278f6b5c3e62834bde</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.SourceLink.GitHub" Version="8.0.0-beta.23218.3">
```
Comment (mmitche) [inline eng/Version.Details.xml:59]: We should exclude these changes initially. Most repos have coherent parent dependency attributes on their sourcelink dependencies. This means that the arcade updates will fail. Instead, you should:
1. Do the changes in `eng/Versions.props` and `RepositoryInfo.targets`
2. Flow this to all repos
3. Remove the sourcelink dependencies in all repos' `Version.Details.xml` 
4. Remove the sourcelink dependencies here.

**Comment by tmat** [comment]
@ViktorHofer Help would be appreciated. Thanks!

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**